### PR TITLE
Controls

### PR DESCRIPTION
**Pull request by jburel**
* Use the physical size for the voxel
* Resize the png, this allows us to view the large image provided by michael

### CODE REVIEW

**Comment by jburel** [comment]
lsmbioimage

File: omero_fpbioimage/views.py [hunk 1]
```diff
@@ -15,9 +16,31 @@ def index(request):
...
+    y = 1
+    z = 1
+
+    if (px is not None):
+        size = image.getPixelSizeX(True)
+        x = size.getValue()
+
+    py = image.getPrimaryPixels().getPhysicalSizeY()
+    if (py is not None):
+        size = image.getPixelSizeZ(True)
```
Comment (will-moore) [inline omero_fpbioimage/views.py]: Should be ```getPixelSizeY``` not Z

File: omero_fpbioimage/views.py [hunk 2]
```diff
@@ -15,9 +16,31 @@ def index(request):
...
-    image = conn.getObject('image', image_id)
-    context = {'image': image}
-
+    image = conn.getObject('Image', image_id)
+    px = image.getPrimaryPixels().getPhysicalSizeX()
+    x = 1
+    y = 1
+    z = 1
+
+    if (px is not None):
```
Comment (will-moore) [inline omero_fpbioimage/views.py]: Don't need brackets on lines like this.

File: omero_fpbioimage/views.py [hunk 3]
```diff
@@ -26,8 +49,13 @@ def fpbioimage_png(request, image_id, the_z, conn=None, **kwargs):
     """Render png for image at specified Z section."""
     image = conn.getObject('image', image_id)
     jpeg_data = image.renderJpeg(the_z, 0, compression=0.9)
-
+    width = max(image.getSizeX, image.getSizeY)
```
Comment (will-moore) [inline omero_fpbioimage/views.py]: You need to call these methods. ```image.getSizeX()```. This is done for you in the template, but not here!

File: omero_fpbioimage/views.py [hunk 4]
```diff
@@ -26,8 +49,13 @@ def fpbioimage_png(request, image_id, the_z, conn=None, **kwargs):
...
+    if width > 500:
+        width = 500
     i = Image.open(StringIO(jpeg_data))
+    p = (width/float(i.size[0]))
+    hsize = int((float(i.size[1])*float(p)))
+    i = i.resize((width, hsize), PIL.Image.ANTIALIAS)
```
Comment (will-moore) [inline omero_fpbioimage/views.py]: If sizeY is greater than sizeX then sizeX remains at 500, or unchanged if less than 500.
Also, p can be greater than 1 (scale up).
E.g. I have an image of ```125 x 484``` and it's being scaled by ```p 3.872``` to ```484 x 1874```.

File: omero_fpbioimage/views.py (hunk 1 above)
Comment (jburel) [reply omero_fpbioimage/views.py]: well spotted

File: omero_fpbioimage/views.py (hunk 4 above)
Comment (jburel) [reply omero_fpbioimage/views.py]: I did things quickly last night (tired) and I only tested with big one to be honest

**Comment by jburel** [comment]
lsm-2

**Pull request merged**